File: mycollab-services/src/main/java/com/mycollab/aspect/Watchable.java
package com.mycollab.aspect;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(value = {ElementType.TYPE })
public @interface Watchable {
	String userFieldName() default "";
	
	String extraTypeId() default "";
}

File: mycollab-services/src/main/java/com/mycollab/aspect/Watchable.kt
package com.mycollab.aspect

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class Watchable(val userFieldName: String = "", val extraTypeId: String = "")